sdcflows/workflows/gre.py

```diff
@@ -0,0 +1,167 @@
+                  name='bet')
+
+    workflow.connect([
+        (inputnode, magmrg, [('magnitude', 'in_files')]),
+        (magmrg, n4_correct, [('out_file', 'input_image')]),
+        (n4_correct, bet, [('output_image', 'in_file')]),
+        (bet, outputnode, [('mask_file', 'fmap_mask'),
+                           ('out_file', 'fmap_ref'),
+                           ('out_report', 'mask_report')]),
+    ])
+    return workflow
+
+
+def init_fmap_postproc_wf(omp_nthreads, fmap_bspline, median_kernel_size=3,
+                          name='fmap_postproc_wf'):
+    """
+    Postprocess a B0 map estimated elsewhere.
+
+    This workflow denoises (mostly via smoothing) a B0 fieldmap.
+
+    Workflow Graph
+        .. workflow ::
+            :graph2use: orig
+            :simple_form: yes
+
+            from sdcflows.workflows.fmap import init_fmap_postproc_wf
+            wf = init_fmap_postproc_wf(omp_nthreads=6, fmap_bspline=False)
+
+    Parameters
+    ----------
+    omp_nthreads : int
+        Maximum number of threads an individual process may use
+    fmap_bspline : bool
+        Whether the fieldmap should be smoothed and extrapolated to off-brain regions
+        using B-Spline basis.
+    median_kernel_size : int
+        Size of the kernel when smoothing is done with a median filter.
+    name : str
+        Name of workflow (default: ``fmap_postproc_wf``)
+
+    Inputs
+    ------
+    fmap_mask : pathlike
+        A brain binary mask corresponding to this fieldmap.
+    fmap_ref : pathlike
+        A preprocessed magnitude/reference image for the fieldmap.
+    fmap : pathlike
+        A B0-field nonuniformity map (aka fieldmap) estimated elsewhere.
+
+    Outputs
+    -------
+    out_fmap : pathlike
+        Postprocessed fieldmap.
+
+    """
+    workflow = Workflow(name=name)
+    inputnode = pe.Node(niu.IdentityInterface(
+        fields=['fmap_mask', 'fmap_ref', 'fmap']), name='inputnode')
+    outputnode = pe.Node(niu.IdentityInterface(fields=['out_fmap']),
+                         name='outputnode')
+    if fmap_bspline:
+        # despike_threshold=1.0, mask_erode=1),
+        fmapenh = pe.Node(
+            FieldEnhance(unwrap=False, despike=False),
+            name='fmapenh', mem_gb=4, n_procs=omp_nthreads)
+
+        workflow.connect([
+            (inputnode, fmapenh, [('fmap_mask', 'in_mask'),
+                                  ('fmap_ref', 'in_magnitude'),
+                                  ('fmap_hz', 'in_file')]),
+            (fmapenh, outputnode, [('out_file', 'out_fmap')]),
+        ])
+
+    else:
+        denoise = pe.Node(fsl.SpatialFilter(operation='median', kernel_shape='sphere',
+                                            kernel_size=median_kernel_size), name='denoise')
+        demean = pe.Node(niu.Function(function=demean_image), name='demean')
+        cleanup_wf = cleanup_edge_pipeline(name='cleanup_wf')
+        applymsk = pe.Node(fsl.ApplyMask(), name='applymsk')
```

Comment:
This node is new. Why do you want to apply a mask? This is going to exacerbate the problems we already have at the brain boundaries.